hrm-domain/hrm-service/src/profile/profile-routes-v0.ts

```diff
@@ -181,17 +181,19 @@ profilesRouter.post(
     try {
       const { accountSid } = req;
       const { profileId, profileFlagId } = req.params;
-      const { validUntil } = req.query;
+      const { validUntil } = req.body;
 
-      if (validUntil && !isValid(validUntil)) {
+      const parsedValidUntil = parseISO(validUntil);
+
+      if (validUntil && !isValid(parsedValidUntil)) {
```

Comment:
Note that the reason for the tests failing is this:
> when an invalid "validUntil" date is sent, returns 400
Expected: 400
Received: 200

The tests are checking that an "invalid date" is not accepted, and it looks like they are.
You should either validate before parsing or find what happens when you `parseISO( <invalid date> )` and determine what is the result of a parsed invalid date.